### PR TITLE
Refactor System.Diagnostics.FileVersionInfo with PAL

### PR DESCRIPTION
**Pull request by stephentoub**
Introduce a Windows and Unix build for System.Diagnostics.FileVersionInfo.dll.  Windows-specific functionality has been moved to separate files, with Unix-specific (but stubbed out) files added to match.

### CODE REVIEW

File: src/System.Diagnostics.FileVersionInfo/src/System/Diagnostics/FileVersionInfo.cs [hunk 1]
```diff
@@ -349,178 +261,25 @@ public string ProductVersion
...
-                                    }
-                                }
-                            }
-                        }
-                    }
-                }
-            }
-
+            var versionInfo = new FileVersionInfo(fileName);
+            versionInfo.Initialize();
```
Comment (terrajobst) [inline src/System.Diagnostics.FileVersionInfo/src/System/Diagnostics/FileVersionInfo.cs]: Any reason that's not in the .ctor? Seems more logical.

Comment (stephentoub) [inline src/System.Diagnostics.FileVersionInfo/src/System/Diagnostics/FileVersionInfo.cs]: I'll move it.

**Comment by terrajobst** [comment]
Otherwise, LGTM.

**Comment by ellismg** [comment]
test this please

**Comment by ellismg** [comment]
My assumption from @terrajobst comment is that the .ctor is going to become part of the PAL?  If so, LGTM as well.  If not, I'd like to see what you do instead :-).

**Comment by stephentoub** [comment]
@ellismg, thanks.  I just moved the Initialize call into the ctor, such that the check in the ctor didn't have to be duplicated.  If you'd prefer to see the ctor moved in, let me know.

**Comment by terrajobst** [comment]
If the ctor always calls initialize then adding a new PAL forces providing an initialize method. Moving the ctor to the PAL does the same, so I'm fine either way.

**Comment by ellismg** [comment]
@stephentoub I have a slight preference for just putting the .ctor in the PAL.  (This mirrors what I have done internally with some other components).

**Comment by stephentoub** [comment]
@ellismg, done.

**Comment by ellismg** [comment]
@stephentoub Thanks.  LGTM!

**Comment by stephentoub** [comment]
Thanks, @terrajobst and @ellismg.

**Pull request merged**